### ISSUE TITLE
New model: Unemployment search

### ISSUE THREAD

**Issue by mnwhite** [econ-ark/HARK#21]
Create module for models of consumption-saving-search.  Consumption-saving model with persistent unemployment; agent must exert costly effort to increase probability of finding job / receiving offer.

Lots of variations on this to do.  Can pick one or two easy ones to start to show models with multiple controls in HARK (or rather: multiple controls in _some_ states).

**Comment by mnwhite** [comment]
This is a good candidate for an early outside contributor who wants to write a new model.

Consider the utility form: u(c,s) = ((1-s)^alpha*c)^(1-rho))/(1-rho), where c is consumption and s is search effort (0 when employed); alpha and rho are parameters > 0.

**Comment by giang-nghg** [comment]
@mnwhite I'd like to help, but I only have experience in Python, not scientific modeling. Could you explain to me the general idea of how to implement a new model? Or point me to an example/resource that demonstrate it? Thank you very much.

**Comment by mnwhite** [comment]
If you're an experienced programmer but you don't have graduate-level training in economics, this isn't a good issue to get your to toes wet with HARK.  In the post above, I should have specified "an early outside contributor *who is an economist*".

We have other issues that *are* good places to contribute for someone who knows how to program, but doesn't have a PhD in economics.  We want various numeric tools that have no economic content, but are useful in a wide array of situations.  The most obvious one is #49 to code up the Gauss-Hermite approximation to the normal distribution; I think I even link to Matlab code that can be translated.

I should add other "tool request" issues for non-economists to work on.  E.g. optimization routines.

**Comment by sbenthall** [comment]
I'm an experienced programmer with a PhD that's not in economics; but I think I have enough background to make sense of the consumption shock models and their variations.

As these seem like the canonical models used in HARK so far, I figure I should figure out how to build one before making many other contributions.

If you agree, I'd be happy to take on this ticket as an exercise.

**Comment by mnwhite** [comment]
Writing code to solve a new model is actually the *largest* possible
contribution, but I definitely won't tell you not to try.

**Comment by sbenthall** [comment]
I'm looking at this model that already factors in unemployment:

https://github.com/econ-ark/HARK/blob/master/HARK/ConsumptionSaving/ConsMarkovModel.py#L983-L1126

Can you explain how the model you have in mind for this issue differs from this one?

A costly unemployment search could be modeled as a further reduction of "income" when unemployed.

Or is search voluntary (and action with learned policy, not assumed)?

**Comment by mnwhite** [comment]
Yes, the *intensity* of employment search is a second choice (or control)
variable.  A more intense search increases the probability of becoming
re-employed, but is increasingly costly in terms of *utility*, not money.
It can be modeled as the agent choosing their probability of being
re-employed s_t, with a larger hit to utility from larger values of s_t.
For example, the form I've worked with in the past was something like:

u(c,s) = ((1-s)^\alpha * c)^{1-\rho} / (1 - \rho)

Where alpha is a parameter that governs the disutility of search.  When s_t
= 0, that leading factor is just 1 and the agent loses nothing from
searching.

**Comment by mnwhite** [comment]
I have multiple PhD students working on variations of this model right now.  Submissions are due ~Sept 1, will update.

**Comment by mnwhite** [comment]
Those students in 2020 did not produce workable code. Endogenous unemployment models remain desired in HARK, but it's fallen by the wayside.
